syntax = "proto3";

import "google/api/client.proto";
import "google/api/http.proto";
import "google/api/annotations.proto";

option java_multiple_files = true;
option java_outer_classname = "AddressValidationServiceProto";
option java_package = "com.google.maps.addressvalidation.v1";

service AddressValidation {
    // Validates an address.
    rpc ValidateAddress(ValidateAddressRequest)
      returns (ValidateAddressResponse) {
    option (google.api.http) = {
      post: "/v1:validateAddress"
      body: "*"
    };
  }
}

message ValidateAddressRequest {
  string address = 1;
}

message ValidateAddressResponse {
  string message = 1;
}